src/common.h

```diff
@@ -369,6 +369,24 @@ char *ly_strnchr(const char *s, int c, size_t len);
  */
 int ly_strncmp(const char *refstr, const char *str, size_t str_len);
 
+/**
+ * @brief Get xml namespace prefix or just some substring from @p str.
+ *
+ * The function looking for possible xml namespace prefix at the beginning of the @p str_begin and return it.
+ * If there is some string at the @p str_begin and then follows the prefix, then the string is returned.
+ * If @p str_begin does not contain any prefix, then the entire @p str_begin is returned.
+ * The @p rest can be used as the @p str_begin in the next iteration.
+ *
+ * @param[in] str_begin Begin of the input string.
+ * @param[in] str_end End of the input string. If set to NULL then the @p str_begin must be NULL-terminated string.
+ * @param[out] substr_type Type of substring found. Set to 1 for the xml namespace prefix, otherwise 0.
+ * @param[out] rest String the remainder part located behind the found substring.
+ * If the @p substr_type is set to 1 then the colon character is skipped.
+ * If no string remains, it is set to NULL.
+ * @return A number of bytes of substring starting at position @p str_begin.
```

Comment:
I would reword this to `Length of found substring starting at @p str_begin.`, for example.

Reply:
I think the mention of bytes is more accurate, because the `ly_getutf8` function can have up to 4 bytes for one character.

Reply:
Fair enough, then make it `Number of bytes of found substring starting at @p str_begin.`. Generally, when documenting this way with doxygen, skip articles at the beginning, that is the custom.